compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -428,8 +435,10 @@ impl<'tcx> SizeSkeleton<'tcx> {
                     for field in fields {
                         let field = field?;
                         match field {
-                            SizeSkeleton::Known(size) => {
-                                if size.bytes() > 0 {
+                            SizeSkeleton::Known(size, align) => {
+                                let is_1zst = size.bytes() == 0
+                                    && align.is_some_and(|align| align.bytes() == 0);
```

Comment:
```suggestion
                                    && align.is_some_and(|align| align.bytes() == 1);
```

Reply:
Ah, good point... so apparently we have no tests actually covering this.